Infrastructure/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Application.Interfaces.IRepositories;
+using Domain.Models;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class, IEntity
+{
+    protected DataContext _context;
+    private readonly DbSet<T> _dbSet;
+    private readonly ILogger _logger;
+
+    public GenericRepository(DataContext context, ILogger logger)
+    {
+        _context = context;
+        this._dbSet = context.Set<T>();
+        _logger = logger;
+    }
+
+    public virtual async Task<IEnumerable<T>> GetAll()
+    {
+        return await _dbSet.ToListAsync();
+    }
+
+    public virtual async Task<T> GetById(int id)
+    {
+        try
+        {
+            return await _dbSet.FirstOrDefaultAsync(e => e.Id == id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error in GetById method with id: {id}", id);
+            throw new Exception("Error in GetById, {e}", e);
```

Comment:
You don't need to create a new exception, just propagate an existed one, in the stack you'll see that in happened in the GetById method, don't need to add this message. More than that, adding `e` as a string, you hide all stack info. Right way to re-throw an exception just `throw;` instead of `throw new Exception("Error in GetById, {e}", e);`

Reply:
fixed